packages/mobile/src/onboarding/contacts/ImportContactsScreen.tsx

```diff
@@ -1,227 +1,222 @@
+import { useDispatch, useSelector } from 'react-redux'
 import CeloAnalytics from 'src/analytics/CeloAnalytics'
 import { CustomEventNames } from 'src/analytics/constants'
-import { Namespaces, withTranslation } from 'src/i18n'
+import i18n, { Namespaces } from 'src/i18n'
 import LoadingSpinner from 'src/icons/LoadingSpinner'
 import { cancelImportContacts, denyImportContacts, importContacts } from 'src/identity/actions'
-import { ImportContactProgress } from 'src/identity/reducer'
-import { ContactMatches, ImportContactsStatus } from 'src/identity/types'
-import { nuxNavigationOptions } from 'src/navigator/Headers'
+import { importContactsProgressSelector, matchedContactsSelector } from 'src/identity/reducer'
+import { ImportContactsStatus } from 'src/identity/types'
+import { HeaderTitleWithSubtitle, nuxNavigationOptionsNoBackButton } from 'src/navigator/Headers.v2'
 import { navigate } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
-import { RootState } from 'src/redux/reducers'
+import { TopBarTextButton } from 'src/navigator/TopBarButton.v2'
+import { StackParamList } from 'src/navigator/types'
 import { requestContactsPermission } from 'src/utils/permissions'
 
-interface StateProps {
-  importContactsProgress: ImportContactProgress
-  matchedContacts: ContactMatches
-}
-
-interface DispatchProps {
-  importContacts: typeof importContacts
-  cancelImportContacts: typeof cancelImportContacts
-  denyImportContacts: typeof denyImportContacts
-}
-
-type Props = StateProps & DispatchProps & WithTranslation
-
-interface State {
-  isFindMeSwitchChecked: boolean
-}
-
-const mapStateToProps = (state: RootState): StateProps => {
-  return {
-    importContactsProgress: state.identity.importContactsProgress,
-    matchedContacts: state.identity.matchedContacts,
-  }
-}
-
-const mapDispatchToProps = {
-  importContacts,
-  cancelImportContacts,
-  denyImportContacts,
-}
-
-class ImportContactScreen extends React.Component<Props, State> {
-  static navigationOptions = nuxNavigationOptions
-
-  state: State = {
-    isFindMeSwitchChecked: true,
-  }
-
-  onFinishTimeout: number | null = null
+type ScreenProps = StackScreenProps<StackParamList, Screens.ImportContacts>
+type Props = ScreenProps
+
+function ImportContactsScreen({ route, navigation }: Props) {
+  const [isFindMeSwitchChecked, setFindMeSwitch] = useState(true)
+  const { t } = useTranslation(Namespaces.onboarding)
+  const importContactsProgress = useSelector(importContactsProgressSelector)
+  const importStatus = importContactsProgress.status
+  const totalContacts = importContactsProgress.total
+  const prevImportContactProgressRef = React.useRef(importContactsProgress)
+  const matchedContacts = useSelector(matchedContactsSelector)
+  const matchedContactsCount = Object.keys(matchedContacts).length
+  const dispatch = useDispatch()
+  const insets = useSafeArea()
+  const headerHeight = useHeaderHeight()
+
+  function renderStatusContainer() {
```

Comment:
Nit: per our convention, these should be lambdas. Though I would consider pulling this out entirely into its own SFC at the top of the file.